### PR TITLE
norminette: init at 3.3.51

### LINKED ISSUE #205010: norminette (42 schools)
check if a C project is formatted according to "the norm" specified in the provided PDF

* homepage URL: https://github.com/42School/norminette
* source URL: https://github.com/42School/norminette
* license: MIT
* platforms: 

also see this project that ?packages norminette for nix:
https://github.com/lightdiscord/nix-norminette

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

https://github.com/42School/norminette

Closes #205010.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tjni** [comment]
Result of `nixpkgs-review pr 205188` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>norminette</li>
  </ul>
</details>

File: pkgs/development/tools/norminette/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchFromGitHub {
+    owner = "42School";
+    repo = pname;
+    rev = version;
+    hash = "sha256-JpWziUKZPOD+AwiYeHR7e0B9l3XKNNp+XQkZEvynKGY=";
+  };
+
+  checkInputs = with python3Packages; [
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/tools/norminette/default.nix:18]:
```suggestion
    pytestCheckHook
```

File: pkgs/development/tools/norminette/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  ];
+
+  checkPhase = ''
+    PYTHONPATH=norminette pytest
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/tools/norminette/default.nix:23]:
```suggestion
  preCheck = ''
    export PYTHONPATH=norminette
  '';
```

File: pkgs/development/tools/norminette/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = with lib; {
+    description = "Open source norminette to apply 42's norme to C files";
+    homepage = "https://github.com/42School/norminette";
+    license = licenses.mit;
+    maintainers = with maintainers; [ wegank ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/norminette/default.nix:30]:
```suggestion
```
default for buildPythonApplication

File: pkgs/development/tools/norminette/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/tools/norminette/default.nix]: Done. Sorry again for the force push...

**Pull request merged**